tests/test_interpolation.py

```diff
@@ -494,42 +536,43 @@ def test_resolver_dot_start(restore_resolvers: Any) -> None:
         (
             lambda *args: args,
             "escape_whitespace",
-            "${my_resolver:cat\\, do g}",
-            ("cat, do g",),
+            "${my_resolver:cat,\\ do g}",
+            ("cat", " do g"),
         ),
         (lambda: "zero", "zero_arg", "${my_resolver:}", "zero"),
     ],
 )
-def test_resolver_that_allows_a_list_of_arguments(
+def test_resolver_that_allows_a_list_of_arguments_legacy(
     restore_resolvers: Any, resolver: Resolver, name: str, key: str, result: Any
 ) -> None:
-    OmegaConf.register_resolver("my_resolver", resolver)
+    OmegaConf.legacy_register_resolver("my_resolver", resolver)
     c = OmegaConf.create({name: key})
     assert c[name] == result
 
 
 def test_resolver_deprecated_behavior(restore_resolvers: Any) -> None:
-    OmegaConf.register_resolver("my_resolver", lambda *args: args)
+    with pytest.warns(UserWarning):
+        OmegaConf.register_resolver("my_resolver", lambda *args: args)
     c = OmegaConf.create(
         {
             "int": "${my_resolver:1}",
             "null": "${my_resolver:null}",
             "bool": "${my_resolver:TruE,falSE}",
             "str": "${my_resolver:a,b,c}",
+            "inter": "${my_resolver:${int}}",
         }
     )
-    with pytest.warns(UserWarning):
-        assert c.int == ("1",)
-    with pytest.warns(UserWarning):
-        assert c.null == ("null",)
-    with pytest.warns(UserWarning):
-        assert c.bool == ("TruE", "falSE")
+    assert c.int == ("1",)
+    assert c.null == ("null",)
+    assert c.bool == ("TruE", "falSE")
     assert c.str == ("a", "b", "c")
+    with pytest.raises(ValueError):
+        c.inter
```

Comment:
can you explain what this is testing?
It's not clear from reading the code.

Reply:
I added comments in 6b19e94, let me know if they clear it up

Reply:
Yes. I missed that the resolved you registered is just returning the input.